### PR TITLE
Fix image scaling on Wear.

### PR DESCRIPTION
**Pull request by yschimke**
Fixes images generated at 999, instead of 1000.

### CODE REVIEW

**Comment by yschimke** [comment]
PaparazziLoggerTest is passing for me locally...

**Comment by yschimke** [comment]
I'm getting these errors with paparazzi 1.0

```
Component 1 width should be a multiple of 2 for colorspace: YUV420J
java.lang.IllegalArgumentException: Component 1 width should be a multiple of 2 for colorspace: YUV420J
	at org.jcodec.common.model.Picture.<init>(Picture.java:54)
	at org.jcodec.common.model.Picture.createCropped(Picture.java:94)
	at org.jcodec.common.model.Picture.create(Picture.java:75)
	at org.jcodec.api.transcode.PixelStoreImpl.getPicture(PixelStoreImpl.java:25)
	at org.jcodec.api.SequenceEncoder.encodeNativeFrame(SequenceEncoder.java:94)
	at org.jcodec.api.awt.AWTSequenceEncoder.encodeImage(AWTSequenceEncoder.java:49)
	at app.cash.paparazzi.HtmlReportWriter.writeVideo(HtmlReportWriter.kt:170)
```

I suspect that neither toInt() or roundToInt() are safe in this case, and we should roundToEven?

**Comment by yschimke** [comment]
Failing on 

```
app.cash.paparazzi.gradle.PaparazziPluginTest > configIsUpdatable FAILED
    java.lang.IllegalArgumentException: Images must have the same dimensions: (626,1000) vs. (625,1000)
```

So 

current - gives 999 images on wear, presumably fails on videos.
roundToInt() - fails on many videos (must be even number of pixels)
this - fails on config test

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt [hunk 1]
```diff
@@ -228,8 +233,14 @@ internal object ImageUtils {
...
     var sourceWidth = source.width
     var sourceHeight = source.height
-    val destWidth = Math.max(1, (xScale * sourceWidth).toInt())
-    val destHeight = Math.max(1, (yScale * sourceHeight).toInt())
+
+    // Avoid errors with jcodec
+    // "Component 1 width should be a multiple of 2 for colorspace: YUV420J"
+    val destWidth = Math.max(1, (xScale * sourceWidth).roundToEven())
+
+    // Round to nearest int, rather than floor.
```
Comment (yschimke) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt:241]: @jrodbx I wonder whether this should round in the same direction as width?

**Comment by yschimke** [comment]
Next

```
app.cash.paparazzi.gradle.PaparazziPluginTest > verifyRenderingModes FAILED
    java.lang.IllegalArgumentException: Images must have the same dimensions: (1000,593) vs. (1000,592)
        at app.cash.paparazzi.gradle.ImageSubject$ImageAssert.getDifferencePercent(ImageSubject.kt:68)
        at app.cash.paparazzi.gradle.ImageSubject$ImageAssert.withThreshold(ImageSubject.kt:52)
        at app.cash.paparazzi.gradle.ImageSubject$ImageAssert.withDefaultThreshold(ImageSubject.kt:40)
        at app.cash.paparazzi.gradle.PaparazziPluginTest.verifyRenderingModes(PaparazziPluginTest.kt:527)
```

**Comment by jrodbx** [comment]
@yschimke is this still an issue, or was this fixed in the other PR by updating the golden image?

if the latter, could you remind me what the actual issue was?

**Comment by yschimke** [comment]
I'll test. We definitely have 999x999 images with 1.0

And they break gif support

**Comment by yschimke** [comment]
@jrodbx added a test to highlight the problem

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt [hunk 2]
```diff
@@ -228,11 +233,18 @@ internal object ImageUtils {
...
     var sourceHeight = source.height
-    val destWidth = Math.max(1, (xScale * sourceWidth).toInt())
-    val destHeight = Math.max(1, (yScale * sourceHeight).toInt())
+
+    // Avoid errors with jcodec
+    // "Component 1 width should be a multiple of 2 for colorspace: YUV420J"
+//    val destWidth = (xScale * sourceWidth).roundToEven().coerceAtLeast(1)
+//    val destHeight = (yScale * sourceHeight).roundToEven().coerceAtLeast(1)
+
+    val destWidth = max(1, (xScale * sourceWidth).toInt())
```
Comment (yschimke) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt:242]: Reverted to see failure

**Comment by yschimke** [comment]
Not reproducible, so possibly fixed with unrelated change.

**Pull request closed without merging**